go-selinux/selinux.go

```diff
@@ -63,8 +63,11 @@ func FileLabel(fpath string) (string, error) {
 	return fileLabel(fpath)
 }
 
-// SetFSCreateLabel tells kernel the label to create all file system objects
-// created by this task. Setting label="" to return to default.
+// SetFSCreateLabel tells the kernel the label to create all file system objects
+// created by this task. Set label="" to return to the default label. Calls to SetFSCreateLabel
+// should be wrapped in runtime.LockOSThread()/runtime.UnlockOSThread() until file system
+// objects created by this task are finished to guarantee another goroutine does not migrate
+// to the current thread before execution is complete.
```

Comment:
Not a blocker (as it was already a bit confusing), but the first line is a bit hard to grasp. Perhaps something like;

```suggestion
// SetFSCreateLabel tells the kernel what label to use for all file system objects created by this task.
// Set label to an empty string to return to the default label. Calls to SetFSCreateLabel
// should be wrapped in runtime.LockOSThread()/runtime.UnlockOSThread() until file system
// objects created by this task are finished to guarantee another goroutine does not migrate
// to the current thread before execution is complete.
```

(suggestions welcome! /cc @rhatdan)